bins/reth-crawler/src/crawler/listener/update_listener.rs

```diff
@@ -303,6 +337,27 @@ impl UpdateListener {
                 let best_block = their_status.blockhash.to_string();
                 let genesis_block_hash = their_status.genesis.to_string();
 
+                // check if peer is synced with the latest chain's blocks
+                let mut synced = None;
+                if let Ok(last_block_number) = provider.get_block_number().await {
+                    info!("last block number: {}", last_block_number);
+                    let peer_best_block_hash = Into::<H256>::into(their_status.blockhash.0);
+                    info!("peer best block hash: {:#?}", peer_best_block_hash);
+                    if let Ok(Some(peer_best_block)) =
+                        provider.get_block(peer_best_block_hash).await
```

Comment:
if this call fails for w/e reason, we just skip setting this flag altogether is how im reading this

that should be fine, but how do we handle storing a missing flag in SQLite?

Reply:
we handle it right now.

If you see the Sqlite db I did not put `NOT NULL`. So if we do not have this info, sqlite simply displays `NULL`.